DependencyInjection/EkinoNewRelicExtension.php

```diff
@@ -43,52 +45,65 @@ public function load(array $configs, ContainerBuilder $container): void
         $loader = new Loader\XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.xml');
 
-        if (!$config['enabled']) {
-            $config['monolog']['enabled'] = false;
-            $interactor = BlackholeInteractor::class;
-        } elseif (isset($config['interactor'])) {
-            $interactor = $config['interactor'];
-        } else {
-            // Fallback to see if the extension is loaded or not
-            $interactor = \extension_loaded('newrelic') ? NewRelicInteractor::class : BlackholeInteractor::class;
-        }
+        $container->setAlias(NewRelicInteractorInterface::class, $this->getInteractorServiceId($config));
+        $container->setAlias(TransactionNamingStrategyInterface::class, $this->getTransactionNamingServiceId($config));
 
         if ($config['logging']) {
-            $container->getDefinition(LoggingInteractorDecorator::class)
-                ->replaceArgument(0, new Reference($interactor));
-            $interactor = LoggingInteractorDecorator::class;
+            $container->register(LoggingInteractorDecorator::class)
+                 ->setDecoratedService(NewRelicInteractorInterface::class)
+                ->setArguments(
+                    [
+                        '$interactor' => new Reference(LoggingInteractorDecorator::class.'.inner'),
+                        '$logger' => new Reference('logger'),
+                    ]
+                )
+                ->setPublic(false)
+            ;
         }
-        $container->setAlias('ekino.new_relic.interactor', $interactor);
 
         if (!empty($config['deployment_names'])) {
             $config['deployment_names'] = \array_values(\array_filter(\explode(';', $config['application_name'])));
         }
 
         $container->getDefinition(Config::class)
-            ->replaceArgument(0, $config['application_name'])
-            ->replaceArgument(1, $config['api_key'])
-            ->replaceArgument(2, $config['license_key'])
-            ->replaceArgument(3, $config['xmit'])
-            ->replaceArgument(4, $config['deployment_names'])
-        ;
+            ->setArguments(
+                [
+                    '$name' => $config['application_name'],
+                    '$apiKey' => $config['api_key'],
+                    '$licenseKey' => $config['license_key'],
+                    '$xmit' => $config['xmit'],
+                    '$deploymentNames' => $config['deployment_names'],
```

Comment:
Will this mean that if we change the variable name in the `Config` class then this declaration will break?

Reply:
indeed, but since symfony introduced the `named argument` feature, changing the name of an argument in a constructor (or a mutator) should be considered as a BC break